File: mybin/se/sysconfig/formatter/cs/examples/forstmt.cs
public class Statement {
   private void traverse(Node root, List<int> numbers) {
      int ct;
      Node n;

      for (ct = 0, n = root; n != null; ct++, n = n.next()) {
         do_something(n);
      }

      foreach (int num in numbers) {
         do_something(num);
      }
   }
}

File: mybin/se/sysconfig/formatter/m/examples/forstmt.m
void traverse(id* handler, Node *root, std::vector<int> numbers)
{
    int ct;
    Node *n;

    for (ct = 0, n = root; n != null; ct++, n=n->next()) {
       [handler doSomethingWith: n];
    }
	[handler bork];
}
